src/sync/installation.ts

```diff
@@ -349,18 +349,22 @@ const findSubscriptionForMessage = (data: BackfillMessagePayload) =>
 		data.gitHubAppConfig?.gitHubAppId
 	);
 
-export const markCurrentTaskAsFailedAndContinue = async (data: BackfillMessagePayload, nextTask: Task, isPermissionError: boolean, sendBackfillMessage: (message, delay, logger) => Promise<unknown>, log: Logger): Promise<void> => {
+export const markCurrentTaskAsFailedAndContinue = async (data: BackfillMessagePayload, nextTask: Task, isPermissionError: boolean, sendBackfillMessage: (message, delay, logger, err: Error) => Promise<unknown>, log: Logger): Promise<void> => {
 	const subscription = await findSubscriptionForMessage(data);
 	if (!subscription) {
 		log.warn("No subscription found, nothing to do");
 		return;
 	}
 
+	// marking the current task as failed, this value will override any preexisting failedCodes and only keep the last known failed issue.
+	const failedCode = getFailedCode(err);
+
 	// marking the current task as failed
-	await updateRepo(subscription, nextTask.repositoryId, { [getStatusKey(nextTask.task)]: "failed" });
+	await updateRepo(subscription, nextTask.repositoryId, { [getStatusKey(nextTask.task)]: "failed", failedCode });
```

Comment:
Can you confirm when the nextTask.task === 'repository', this line won't break? 
ps: I check in my local, seems fine though. Just Fyi.

Reply:
Yup, this is fine, it was already updating the repotable its just also setting failedCode